### PR TITLE
feat(cohorts): Allow adding property filters to events in cohorts

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

I'm not super happy with how things are here, but trying to refactor everything was a much bigger chunk of work 😅 .

The problem: I want to create a cohort of exposed users for experiments automatically. This requires:

(1) Adding filters to events in cohorts. The action workaround is too annoying because this is generated automatically, and more confusing to also generate an action at the same time vs just showing the filters on the cohort. Also, most people don't realise this workaround exists, so its an overall improvement.

(2) Exact dates for cohort calculation. Since the experiment starts on a fixed date, we want to calculate exposed users starting from this date, not a relative date.

## Changes

Ideally, I'd add these above features to all possible cohort types, but that means handling these filters everywhere, and has some complications in the more advanced cases, which becomes a much bigger job, so I opted to add these to only the basic performed event conditions.

The way things are setup, all cohort filters are modelled as part of `Property`, which means we need to add these new props here as well, barring a big refactor.

I combined the `time_unit` and `time_interval` fields into an `explicit_datetime` field which works much like the date range in regular insights. I can't remove the former yet though because there are a lot more places its used.

Adding the filters is straightforward because we _almost_ do the same thing already via the action filters.

image

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by neilkakkar** [comment]
Found a blocking bug to fix: Need to validate criteriaToHumanSentence - it breaks badly

File: frontend/src/scenes/cohorts/CohortFilters/CohortField.tsx [hunk 1]
```diff
@@ -167,6 +173,118 @@ export function CohortPersonPropertiesValuesField({
...
+        onChange: _onChange,
+    })
+    const { value } = useValues(logic)
+    const { onChange } = useActions(logic)
+    const componentRef = useRef<HTMLDivElement>(null)
+
+    const valueExists = ((value as AnyPropertyFilter[]) || []).length > 0
+
+    useEffect(() => {
+        // :TRICKY: We check paremt has CohortCriteriaRow__Criteria__Field class and add basis-full class if value exists
```
Comment (mariusandra) [inline frontend/src/scenes/cohorts/CohortFilters/CohortField.tsx:197]:
```suggestion
        // :TRICKY: We check parent has CohortCriteriaRow__Criteria__Field class and add basis-full class if value exists
```

Indeed, this is pretty hacky though.

**Review by mariusandra — APPROVED** [review]
Feels a bit hacky at times, but I don't see anything blocking... since we'll need to migrate most of this into HogQL eventually.... and as long as you volunteer to quickly fix any issues that come from this in the next days ✅

**Comment by neilkakkar** [comment]
Yep for sure, will fix anything that comes out of this 👍

**Pull request merged**